src/libopensc/card-epass2003.c

```diff
@@ -2444,7 +2445,32 @@ epass2003_gen_key(struct sc_card *card, sc_epass2003_gen_key_data * data)
 	if (!data->modulus)
 		LOG_FUNC_RETURN(card->ctx, SC_ERROR_OUT_OF_MEMORY);
 
-	memcpy(data->modulus, rbuf, len);
+
+	if(len == 256)
+	{
+		unsigned char tmp[256] = {0};
+		int len1 = 0;
+		int len2 = 0;
+		//get x value
+		if(rbuf[0] == 0x58 )
+		{
+			len1 = rbuf[1];
+			memcpy(&tmp[0],&rbuf[2],len1);
+		}
```

Comment:
If the buffer does not contain the expected data, we will copy just zeroes to the modulus?